### PR TITLE
SRE-2455 / secretstores Chart

### PR DESCRIPTION
**Pull request by pshelby**
JIRA SRE-2455

## Summary
Creating secretstores shared chart for defining namespaced SecretStores.

## Testing
Tested locally with implementation in some-testserver.  Diff shared in Slack.

### CODE REVIEW

File: charts/secretstores/templates/externalsecret.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  secretStoreRef:
+    name: {{$key}}
+    kind: SecretStore
+  target:
+    name: {{$secret.destName}}
+  data:
+    - remoteRef:
+        key: {{$secret.sourcePath}}
+        property: {{$secret.sourceKey}}
+      secretKey: {{$secret.destKey}}
```
Comment (dansc0de) [inline charts/secretstores/templates/externalsecret.yaml:18]: should we support [dataFrom](https://external-secrets.io/v0.4.4/api-externalsecret/) as well?

File: charts/secretstores/templates/externalsecret.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{{- range $key, $value := $.Values.stores}}
+{{- range $index, $secret := $value.secrets}}
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: {{$secret.destName}}
+spec:
+  refreshInterval: {{$secret.refreshInterval | default "1h"}}
```
Comment (keegan-tatari) [inline charts/secretstores/templates/externalsecret.yaml:8]: great, we should start doing this 1hr default everywhere :)

**Review by keegan-tatari — APPROVED** [review]
:rocket:

File: charts/secretstores/templates/externalsecret.yaml (hunk 1 above)
Comment (pshelby) [reply charts/secretstores/templates/externalsecret.yaml:18]: Yeah, good call.  I'll look to see how dataFrom can be scoped down to a subset of a store.

**Review by stephen-tatari — APPROVED** [review]
🎉🚢

File: charts/secretstores/templates/externalsecret.yaml [hunk 3]
```diff
@@ -1,21 +1,54 @@
...
+  labels:
+    {{- include "secretstores.labels" $ | nindent 4 }}
 spec:
-  refreshInterval: {{$secret.refreshInterval | default "1h"}}
+  refreshInterval: {{$value.refreshInterval | default "1h" | quote}}
   secretStoreRef:
-    name: {{$key}}
-    kind: SecretStore
+    name: {{$key | quote}}
+    kind: "SecretStore"
```
Comment (keegan-tatari) [inline charts/secretstores/templates/externalsecret.yaml:12]: no quotes here right?

File: charts/secretstores/templates/secretstore.yaml [hunk 4]
```diff
@@ -1,9 +1,11 @@
 {{- range $key, $value := $.Values.stores}}
-apiVersion: external-secrets.io/v1beta1
-kind: SecretStore
+apiVersion: "external-secrets.io/v1beta1"
+kind: "SecretStore"
```
Comment (keegan-tatari) [inline charts/secretstores/templates/secretstore.yaml:3]: no quotes on kind/apiVersion

File: charts/secretstores/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    # Vault example for service/ mount
+    vault-service:
+      type: "vault"
+      server: "http://cool.vault.path:port"
+      path: "service"
+      secrets:
+        - sourcePath: "my-service/test-secret"
+          sourceKey: "nunya"
+          destName: "test-secret-vault-service"
+          destKey: "nun-ya"
```
Comment (dansc0de) [inline charts/secretstores/README.md:68]: is destKey optional? I think it is only used for `data`  secrets. Is that correct?

**Review by dansc0de — APPROVED** [review]

File: charts/secretstores/README.md (hunk 5 above)
Comment (pshelby) [reply charts/secretstores/README.md:68]: Yeah, `destKey` is optional; it just allows someone to override the key name within the k8s secret.  I'll mark as optional in Caveats.

File: charts/secretstores/templates/externalsecret.yaml [hunk 6]
```diff
@@ -18,7 +18,7 @@ spec:
     - remoteRef:
         key: {{$secret.sourcePath | quote}}
         property: {{$secret.sourceKey | quote}}
-      secretKey: {{$secret.destKey | quote}}
+      secretKey: {{$secret.destKey | default $secret.sourceKey | quote}}
```
Comment (keegan-tatari) [inline charts/secretstores/templates/externalsecret.yaml:21]: :+1:

**Pull request merged**